golang/cosmos/daemon/cmd/root.go

```diff
@@ -453,7 +480,16 @@ func (ac appCreator) appExport(
 	jailAllowedAddrs []string,
 	appOpts servertypes.AppOptions,
 ) (servertypes.ExportedApp, error) {
-	if OnExportHook != nil {
+	swingStoreExportMode, ok := appOpts.Get(gaia.FlagSwingStoreExportMode).(string)
+	if !(ok && allowedSwingSetExportModes[swingStoreExportMode]) {
+		return servertypes.ExportedApp{}, fmt.Errorf(
+			"export mode '%s' is not supported",
+			swingStoreExportMode,
+		)
+	}
+
+	// We don't have to launch VM in case the swing store export is not required
+	if !(swingStoreExportMode == swingsetkeeper.SwingStoreArtifactModeSkip || OnExportHook == nil) {
```

Comment:
@mhofman since `OnExportHook` currently points to `launchVM`, is it okay to bypass this hook in case swing-store export is not needed? Or do you see us adding cosmic side future hook changes?

Reply:
I think that's fine, but I'd like @michaelfig to confirm, especially with his upcoming changed for split brains. Basically export only needs to launch the VM if it will export the swing-store, which is decided by a command line config. Since `OnExportHook` is unconditionally set or not based on the entrypoint used, the only option we have is to ignore it here.

Reply:
Btw, my brain still struggles with distributing negations. 

```suggestion
	if swingStoreExportMode != swingsetkeeper.SwingStoreArtifactModeSkip && OnExportHook != nil {
```